src/App.tsx

```diff
@@ -15,71 +15,93 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string | undefined;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const { selectedGood } = this.state;
+    const clickedGood = event.currentTarget.dataset.good;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    if (selectedGood === clickedGood) {
+      this.setState({ selectedGood: undefined });
+    } else {
+      this.setState({ selectedGood: clickedGood });
+    }
+  };
+
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
+              className="delete ml-3"
+              onClick={() => this.setState({ selectedGood: undefined })}
             >
-              -
+              <span className="visually-hidden">Clear selected good</span>
             </button>
-          </td>
+          </h1>
+        ) : (
+          <h1 className="title">No goods selected</h1>
+        )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={selectedGood === good
+                  ? 'has-background-success-light' : ''}
+              >
+                <td>
+                  <button
+                    data-good={good}
+                    data-cy="AddButton"
+                    type="button"
+                    className="button"
+                    onClick={this.handleButtonClick}
```

Comment:
Default approach in React, it's pass `good` here, so:
```suggestion
                    onClick={() => this.handleButtonClick(good)}
```
In such way, you won't need `data-good`